src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -184,13 +174,14 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
     }
   }, [selectedDataView, dispatchSidebarStateAction, selectedDataViewRef]);
 
+  const querySubscriberResult = useQuerySubscriber({ data });
   const { isProcessing, refetchFieldsExistenceInfo } = useExistingFieldsFetcher({
     disableAutoFetching: true,
     dataViews: !isPlainRecord && sidebarState.dataView ? [sidebarState.dataView] : [],
-    query: query!,
-    filters,
-    fromDate: sidebarState.dateRange?.fromDate ?? null, // existence fetching will be skipped if `null`
-    toDate: sidebarState.dateRange?.toDate ?? null,
+    query: querySubscriberResult.query,
+    filters: querySubscriberResult.filters,
+    fromDate: querySubscriberResult.fromDate,
+    toDate: querySubscriberResult.toDate,
```

Comment:
@kertal Updated after https://github.com/elastic/kibana/pull/145332
Now it handles changes to query, filters (app/global), date range (based on session state) in one place.